### ISSUE TITLE
Worm Segmentation Network on MIPs

### ISSUE THREAD

**Issue by tibuch** [fmi-basel/ggrossha_SWI#7]
The worm segmentation could be more robust by training and applying it to the MIPs only. Currently the input to the network are 25 z-planes treated as 25 input channels and the network produces a 2D segmentation map. To train on MIPs it would only be necessary to change the number of input channels to 1 and ensure that the training data is prepared correctly.

Points of Interest:
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/workflow.nf#L107 change the `SEGMENT` process input to `projections` to ensure that all MIPs are there. 
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/prepare_annotation_data.py#L47-L49 use the computed MIPs for the micro-sam embedding computation.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/prepare_train_data.py#L77 use `mips` instead of `0`.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/run.py#L41-L43 use `mips` instead of `0`.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/run.py#L55-L60 do __not__ pad.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/run.py#L130-L132 can be removed.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/network.py#L57-L64 remove padding.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/network.py#L181 `in_channels = 1`.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/train_config.py#L36 it might be possible to increase the `batch_size`.
* https://github.com/fmi-basel/ggrossha_SWI-Template/blob/c1e9cd9722fef02e4588ba2736bbebc421e2721e/source/s02_segment/train_config.py#L103-L114 not necessary anymore.

That should be it, but most likely I missed some lines along the way.

**Comment by imagejan** [comment]
The way how `mips` are currently written doesn't produce a valid OME-Zarr array (a zarr array is created in the `mips` subfolder, but its metadata aren't written to the `.zattrs` of the containing group). This causes issues when preparing training data from the `.zarr` datasets that contain both the original 3D array (at path `0`) and the projection (at path `mips`).

A clean solution would be to write the projections into a separate zarr folder, or to properly add metadata for the additional group.

To avoid future trouble with maintenance of the code for writing the arrays etc., I suggest we migrate to using `ngio` for writing and reading valid OME-Zarr files.
